core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1831,6 +1834,11 @@ private void commitIndexWriter(final IndexWriter writer, final Translog translog
         }
     }
 
+    // commit hook for testing
+    void callCommitOnWriter(IndexWriter writer) throws IOException {
```

Comment:
I struggled with this name. We already have a `commitIndexWriter`, so I figured to be explicit and call this exactly what this does.